### ISSUE TITLE
Capability Delegation

### ISSUE THREAD

**Issue by mustaqahmed** [mozilla/standards-positions#565]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: Capability Delegation
* Specification or proposal URL: https://wicg.github.io/capability-delegation/spec.html
* Mozillians who can provide input (optional): @annevk
* Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1830200

### Other information

* [TAG review](https://github.com/w3ctag/design-reviews/issues/655) is ongoing.
* Note that we moved away from our past attempt to delegate "user activation", which turned out to be too generic ([previous TAG request](https://github.com/w3ctag/design-reviews/issues/347)).

**Comment by annevk** [comment]
I'm not a 100% sure I understand this correctly (I filed some issues) but I assume the idea is that a parent that has user activation can forward that user activation to a child, but tie it to a particular feature to ensure the child doesn't use it for something else. So the user clicks in the parent and the parent wants the child to go fullscreen upon a click so they send a message to the child allowing it to go fullscreen. This consumes the click in the parent and the child now has the ability to go fullscreen if it so desires.

I agree that seems useful.

Coupling it with messaging seems reasonable. There's no other communication channel available to relay such a thing.

**Comment by mustaqahmed** [comment]
Thanks @annevk, your first para is essentially the proposal here in simple term.  Looking forward to your ideas and suggestions in the WICG repo.

**Comment by hsinyi** [comment]
blink-dev's Intent to Implement and Ship https://groups.google.com/a/chromium.org/g/blink-dev/c/7YkubntWi3Y/m/BKgFpsLYBQAJ That intent mentioned Gecko's signal which doesn't sound 100% accurate, and Mike Taylor has pointed that out.

There are several open issues filed by @annevk under https://github.com/WICG/capability-delegation/issues/ Does that mean the official position wouldn't be made before they are resolved?

**Comment by mustaqahmed** [comment]
Adding a note to the last comment above: we have two intents on blink-dev to cover this:
- Generic mechanism plus Payment Request capability: [Intent to Ship](https://groups.google.com/a/chromium.org/g/blink-dev/c/PHT_2X7oRBE/m/gR9UiZxBAQAJ)
- Fullscreen capability: [Intent to Implement and Ship](https://groups.google.com/a/chromium.org/g/blink-dev/c/7YkubntWi3Y/m/5NdQPDAqBAAJ)

**Comment by annevk** [comment]
I think with the reframing discussed in https://github.com/WICG/capability-delegation/issues/9#issuecomment-1127845526 whereby this essentially extends Permissions Policy's "allowed to use" with an option to also check (delegated) user activation this will be in a good shape, with a clear model for web developers and specification authors.

At that point I'd recommend marking this as worth prototyping.

**Comment by jan-ivar** [comment]
I agree with @annevk here, and think we can mark this as "positive".

**Comment by jan-ivar** [comment]
We'd still like to see https://github.com/WICG/capability-delegation/issues/9 fixed, even if it's largely editorial.

**Comment by martinthomson** [comment]
I think that we should defer until https://github.com/WICG/capability-delegation/issues/9 is resolved.  I would have expected that issue to have been resolved in the last 9 months, but I see no movement.

I would also prefer that this be integrated into permissions policy, rather than live as a standalone specification.

**Comment by mustaqahmed** [comment]
Yes, we (Chrome) want to resolve WICG/capability-delegation#9 and other outstanding issues there so that at least our proposed [HTML monkey patch](https://wicg.github.io/capability-delegation/spec.html#monkey-patch-to-html-initiating-delegation) can gracefully land sometime this year!  We are currently looking into prioritizing this work.

**Comment by mustaqahmed** [comment]
FYI: we have added this as one of our (Chromium/Interactions team) Q2 goals, hopefully we will be able to resolve all blockers to land the HTML spec.

**Comment by zcorpan** [comment]
@mustaqahmed what is the current status and plan for the open issues and upstreaming?

**Comment by zcorpan** [comment]
Suggest positive since it's needed for some use cases where `document.domain` is currently used (e.g. requesting fullscreen across same-site but cross-origin frames).
